marketo/url.go

```diff
@@ -0,0 +1,35 @@
+package marketo
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/amp-labs/connectors/common"
+	"github.com/amp-labs/connectors/common/urlbuilder"
+)
+
+var restAPIPrefix string = "rest" //nolint:gochecknoglobals
+
+func (c *Connector) getURL(params common.ReadParams) (*urlbuilder.URL, error) {
+	// make sure the object is in lowercase format.
+	objName := strings.ToLower(params.ObjectName)
+
+	bURL := strings.Join([]string{c.BaseURL, restAPIPrefix, c.Module, objName}, "/")
+	bURL += ".json"
```

Comment:
Can you add a check to make sure that the base URL already doesn't have a `.json` suffix? It's possible that a connector user might consider the entire name (`leads.json`) as the object name, so we should plan for that.